blueprints/server-capabilitites.md

```diff
@@ -0,0 +1,162 @@
+
+Initially, this is completely backwards-compatible on upgrade, because initially no Delivery Service will have any Required Capabilities.
+
+This feature will be completely optional. TC operators who don't need Server Capabilities will simply not create them.
+
+Server Capabilities will be arbitrary strings. The ATS project will not "seed" any, and impose as little direction as possible. For example, Server Capabilitites could be Cache types, Server types, Hardware types, ATS versions, Lua support, other ATS plugin support, or any other features an operator needs to route (or not route) on.
+
+
+## Proposed Change
+
+- Servers have Capabilities (i.e.: CACHE_MEMORY, CACHE_DISK )
+
+- Delivery Services have Required Capabilities (i.e.: CACHE_MEMORY, CACHE_DISK)
+
+- When generating Configuration (e.g. ATS parent.config):
+  - If a Mid Server which is otherwise parented to an Edge does not have all Required Capabilities of a Delivery Service, that Mid will not be inserted as a parent for that Delivery Service’s remap and parent rules.
+  - Backward Compatibility is automatic:
+    - Initially, no Delivery Services have Required Capabilities. Hence, everything behaves like it does today.
+    - If an EDGE server does not have all capabilities required by a DS, that server SHOULD NOT be assignable to that DS. 
+    - If an EDGE server is assigned to a DS which requires capabilities that server has, it SHOULD NOT be possible to remove those capabilities from that server.
+
+### Traffic Portal Impact
+
+- Server Page
+  - New dropdown to add to a text list: Capabilities
+  - List elements have a button to remove the capability
+  - Dropdown/List is just one option - other GUI components may be used
+  - Component should consider that there could be many Capabilities
+
+- Delivery Service Page
+  - New dropdown to add to a text list: Required Capabilities
+  - List elements have a button to remove a capability
+  - Dropdown/List is one option - other GUI components may be used
+  - Component should consider that there could be many Capabilities
+
+- New page for Server Capability Types
+  - Ability to add, delete Server Capability Types
+  - May be a text input, which adds to a list, with list element buttons to remove from the list
+
+
+### Traffic Ops Impact
+
+`/server_server_capabilities`
+  - List all server capabilities assigned to a server
+  - GET+POST+DELETE
+  - PUT doesn't make sense. To remove one and add another, DELETE and POST.
+
+`/deliveryservices_required_capabilities`
+  - List all DS Required Capabilities
+  - GET+POST+DELETE
+  - PUT doesn't make sense. To remove one and add another, DELETE and POST.
+
+`/server_capabilities`
```

Comment:
readonly users or above for the GET?
admin users only for the POST/DELETE?

Reply:
Ops rather than Admin, I'd think.

I wasn't sure how much detail was necessary in a blueprint. I figured the PR Review could handle things like that, unless there's a big disagreement, and then I'd think it'd only go here after consensus on the list.